aea/test_tools/test_cases.py

```diff
@@ -464,7 +464,8 @@ def terminate_agents(
             try:
                 process.wait(timeout=timeout)
             except subprocess.TimeoutExpired:
-                process.kill()
+                process.terminate()
+                process.wait(timeout=timeout)
```

Comment:
we opt to terminate instead of kill. .
- on POSIX `.terminate` allows the process to gracefully shutdown, whereas kill is for forceful shutdown and cannot be ignored.
- `subprocess.kill` and `subprocess.terminate` are identical on Windows

We wait for termination, if no `returncode` is set before the timeout a `subprocess.TimeoutExpired` will be raised.

Reply:
I would suggest we keep both logics, first we try killing a process gracefully and if that fails we try killing it forcefully using `os.kill`

Reply:
yes that is indeed better. We can filter exit codes for SIGINT and SIGTERM (and CTRL_C, CTRL_BREAK) from `teardown_class` error logging and report only when `kill` was needed.